client-app/src/pages/account/components/invoices-filter/invoices-filter.ts

```diff
@@ -3,97 +3,89 @@ import Component from "vue-class-component";
 import { LocaleMessages, TranslateResult } from "vue-i18n";
 import { Prop } from "vue-property-decorator";
 import i18n from "@i18n";
-import { InvoicesListFilters } from "@account/store/modules/invoices-list/types";
+import { IPaymentSearchCriteria } from '@common/api/api-clients';
 import { locale } from "@common/constants";
 
 @Component
 export default class InvoicesFilter extends Vue {
   @Prop()
-  invoicesFilter!: InvoicesListFilters;
+  searchCriteria!: IPaymentSearchCriteria;
 
   @Prop()
   availableInvoicesStatuses!: string[];
 
-  startDate: Date | undefined = undefined;
-
-  endDate: Date | undefined = undefined;
-
-  isDateValid: boolean | null = null;
-
   allStatusesSelected = false;
 
-  selectedStatuses: string[] = [];
+  isDateValid: boolean | null = null;
 
   datepickerLabels: LocaleMessages | {} = {};
 
   locale = locale;
 
-  emitChanges(updatedFilters: InvoicesListFilters) {
-    this.$emit("filtersChanged", updatedFilters);
+  emitChanges(searchCriteria: IPaymentSearchCriteria) {
+    this.$emit("searchCriteriaChanged", searchCriteria);
   }
 
   mounted() {
     this.setDatepickerLocalization();
   }
 
-  dateChanged() {
-    if (this.startDate && this.endDate) {
-      this.isDateValid = this.startDate <= this.endDate;
+  dateChanged(startDate?: Date, endDate?: Date) {
+    if (startDate && endDate) {
+      this.isDateValid = startDate <= endDate;
       if (this.isDateValid) {
-        const updatedFilters = { ...this.invoicesFilter, startDate: this.startDate, endDate: this.prepareEndDate() };
-        this.emitChanges(updatedFilters);
+        const searchCriteria = { ...this.searchCriteria, startDate, endDate };
+        this.emitChanges(searchCriteria);
       }
```

Comment:
What will happen if this.isDateValid is invalid? I guess it is also the changes for emitting :)
In difficult if-else scenarios really helpful to add empty else blocks if() {...} else { // idle  - it means I checked it and showed to others that it is ok in this case }

Reply:
Out of scope. We will refactor filters any way, so I don't think we even need to create separate issue for that